crates/solvers/src/infra/config/dex/oneinch/file.rs

```diff
@@ -0,0 +1,80 @@
+use {
+    crate::{
+        domain::eth,
+        infra::{config::dex::file, contracts, dex::oneinch},
+        util::serialize,
+    },
+    serde::Deserialize,
+    serde_with::serde_as,
+    std::path::Path,
+};
+
+#[serde_as]
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+struct Config {
+    /// Chain ID used to automatically determine the address of the settlement
+    /// contract and for metrics.
+    #[serde_as(as = "serialize::ChainId")]
+    chain_id: eth::ChainId,
+
+    /// The URL endpoint for the 1inch API.
+    #[serde_as(as = "Option<serde_with::DisplayFromStr>")]
+    endpoint: Option<reqwest::Url>,
+
+    /// The 1Inch liquidity sources to consider when swapping.
+    #[serde(flatten)]
+    liquidity: Liquidity,
```

Comment:
The `Liquidity` enum is a nice idea but according to the serde [docs](https://serde.rs/container-attrs.html#deny_unknown_fields) `deny_unknown_fields` doesn't play nice with `flatten`.